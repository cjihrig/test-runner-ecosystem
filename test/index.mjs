import { describe, it } from 'node:test';
import { expect } from 'chai';

describe('a suite', () => {
  it('passing test', () => {
    expect(true).to.be.a('boolean').that.equals(true);
  });

  it('failing test', () => {
    expect({foo: 'hello'})
      .to.be.an('object')
      .that.has.property('foo')
      .that.is.a('number', 'should be a number')
      .that.equals(123);
  });

  it.skip('a skipped test', () => {
    throw new Error('does not matter');
  });
});

it('a snapshot test', (t) => {
  const value = { foo: 1 };
  value.circular = value;

  t.assert.snapshot(value);
});
