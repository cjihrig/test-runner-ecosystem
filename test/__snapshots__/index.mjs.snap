exports[`a snapshot test 1`] = `
Object {
  "circular": [Circular],
  "foo": 1,
}
`;
